elastic/apm-server
# idxmgmt: reduce scope of config warning

## Motivation/summary

Only log a warning if `apm-server.data_streams.enabled` is specified, and not the broader `apm-server.data_streams.*`. This avoids warnings due to the use of `apm-server.data_streams.wait_for_integration` which continues to be honoured.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/master/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Run `apm-server -e -E apm-server.data_streams.wait_for_integration=true`, check no warnings are logged
2. Run `apm-server -e -E apm-server.data_streams.enabled=true`, check a warning is logged

## Related issues

None
